### ISSUE TITLE
Canary: SUID Sandbox Helper is not configured properly

### ISSUE THREAD

**Issue by carboneater** [a85/POSTMan-Chrome-Extension#8329]
**Describe the bug**
Since Last Canary Update, Postman exits with the following error
`[115032:0414/172455.991116:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /home/carboneater/Downloads/PostmanCanary/app/chrome-sandbox is owned by root and has mode 4755.`

**To Reproduce**
1. Go to PostmanCanary Folder
2. `./PostmanCanary`
3. `[115032:0414/172455.991116:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /home/carboneater/Downloads/PostmanCanary/app/chrome-sandbox is owned by root and has mode 4755.`

**Expected behavior**
Postman Launches

**Screenshots**
```
PostmanCanary/app $ ls -la
total 154308
drwxr-xr-x 5 carboneater carboneater      4096 Apr 13 12:36 .
drwxr-xr-x 3 carboneater carboneater      4096 Apr 13 12:36 ..
-rw-r--r-- 1 carboneater carboneater    178136 Mar 12 07:54 chrome_100_percent.pak
-rw-r--r-- 1 carboneater carboneater    316984 Mar 12 07:54 chrome_200_percent.pak
-rwxr-xr-x 1 carboneater carboneater   5583008 Mar 12 09:13 chrome-sandbox
-rw-r--r-- 1 carboneater carboneater  10426784 Mar 12 06:18 icudtl.dat
-rwxr-xr-x 1 carboneater carboneater    248072 Mar 12 07:01 libEGL.so
-rwxr-xr-x 1 carboneater carboneater   2973840 Mar 12 07:09 libffmpeg.so
-rwxr-xr-x 1 carboneater carboneater   8375384 Mar 12 07:01 libGLESv2.so
-rw-r--r-- 1 carboneater carboneater      1060 Mar 11 00:46 LICENSE
-rw-r--r-- 1 carboneater carboneater   4675743 Mar 12 06:29 LICENSES.chromium.html
drwxr-xr-x 2 carboneater carboneater      4096 Apr 13 12:36 locales
-rw-r--r-- 1 carboneater carboneater     82118 Mar 12 08:03 natives_blob.bin
-rwxr-xr-x 1 carboneater carboneater 115449040 Mar 12 09:12 _PostmanCanary
-rwxr-xr-x 1 carboneater carboneater      8456 Apr 13 12:36 PostmanCanary
drwxr-xr-x 3 carboneater carboneater      4096 Apr 13 12:36 resources
-rw-r--r-- 1 carboneater carboneater   8646888 Mar 12 08:11 resources.pak
-rw-r--r-- 1 carboneater carboneater    280392 Mar 12 09:10 snapshot_blob.bin
drwxr-xr-x 2 carboneater carboneater      4096 Apr 13 12:36 swiftshader
-rw-r--r-- 1 carboneater carboneater    701056 Mar 12 09:11 v8_context_snapshot.bin
-rw-r--r-- 1 carboneater carboneater         6 Mar 12 06:17 version

```

**App information (please complete the following information):**
 - App Type: Native App
 - Postman Version: 7.1.13 Canary (At least, that's what the canary package says, I downloaded a new one minutes ago...)
 - OS: Linux Debian Bullseye (Testing) amd64

**Additional context**
Previous versions worked well.

Why is Postman requiring root? It's installed as my user...

I downloaded a new Canary Package and dropped it over my previous install. Same failure.

**Comment by harryi3t** [comment]
@carboneater The latest version on the canary channel is v7.23.0-canary01.
I just tried downloading the latest version from the [website](https://www.getpostman.com/downloads/canary) and I got this version only
![image](https://user-images.githubusercontent.com/5207331/79308463-4319e500-7f16-11ea-9ab7-8948e73e7603.png)

Can you confirm the version you are using again?

Also, could you help me find the right copy of `Debian Bullseye`? Can I use `debian-bullseye-DI-alpha2-amd64-netinst.iso` from https://cdimage.debian.org/cdimage/bullseye_di_alpha2/amd64/iso-cd/?

**Comment by carboneater** [comment]
Yeah, the package I downloaded is tagged 7.23.0-canary01
Seems like you have an artefact in `app/version` that's stuck at 7.1.13...  
I re-downloaded and reinstalled just for kicks. Same failure.

As for the bullseye image,  
I usually start from a stable image then switch all the `stable` mirrors to `testing`.  
the `netinst` image might work for you,  
but I'm pretty sure `debian-bullseye-DI-alpha2-i386-xfce-CD-1.iso` will save you time as it will come with the XFCE environment pre-installed.  
It won't be cute, but it should work out of the box.

**Comment by twitharshil** [comment]
Hey @carboneater 
You can easily unblock yourself by following these steps and then try to launch the app: -
Run the following command in terminal -
```
Step 1: sudo chown root path/to/Postman/app/chrome-sandbox 
Step 2: sudo chmod 7655 path/to/Postman/app/chrome-sandbox 
```
Postman is built on Electron that comes with Chromium and NodeJs support to build desktop applications. We have recently updated this platform dependency (Electron) in the above canary release. After updating to newer version we got a new Chromium security feature added which is sandboxing. 
To impose this sandboxing, Chromium’s sandbox on Linux machines requires an OS feature called **CLONE_NEWUSER**. Some Linux distributions are built with this feature restricted to privileged users only. You can read more about this here https://github.com/electron/electron/issues/17972#issuecomment-516957971

So when **CLONE_NEWUSER** is unavailable, Chromium uses the **chrome-sandbox** helper binary shipped inside the app. This helper binary needs to perform some functions that are only available to root.
The above steps are giving the root permission and sets the required mode permission ( read, write permissions for user and sets read, execute permissions for Group) for the chrome-sandbox binary file. 
Do try the above step and let us know :)

**Comment by twitharshil** [comment]
Hey @carboneater 
We’ve pushed the updates in the latest canary app that should fix this issue and the same is available in stable app v7.25.1 onwards 🎉 
Feel free to reopen the issue if you are still facing this problem on Postman v7.25.1 onwards.
